mypy/checker.py

```diff
@@ -107,6 +107,10 @@
         ('is_upper_bound', bool),  # False => precise type
     ])
 
+# Keeps track of partial types in a single scope
+PartialTypeScope = NamedTuple('PartialTypeMap', [('map', Dict[Var, Context]),
+                                                 ('is_function', bool)])
```

Comment:
I think a short comment about why we need to keep track about `is_function` will help here. (My understanding this is because fine-grained targets are always top level functions.)